### DISCUSSION TITLE
Can rspack build support the output of gzip files?

### DISCUSSION THREAD

**Discussion by sweetwisdom** [web-infra-dev/rsbuild#2172] [Q&A] [answered]
![image](https://github.com/web-infra-dev/rsbuild/assets/57869305/02aaf36e-e8ba-481d-bc67-acf56982ee61)
目前dev已经支持gzip,希望在build时也能支持输出gzip

**Comment by 9aoy** [comment]
server.compress 在 dev 和 preview 时同样生效，你可以试下 
![image](https://github.com/web-infra-dev/rsbuild/assets/22373761/180c735d-c1ab-4c33-8504-64a7b0100ea8)

**Comment by chenjiahan** [comment]
gzip 是 server 端处理的，不是构建直接生产 gzip 文件

**Comment by xc2** [comment, marked as answer]
可以使用这个 webpack 插件：https://github.com/webpack-contrib/compression-webpack-plugin

```ts
// rsbuild.config.ts
import CompressionPlugin from "compression-webpack-plugin";


export default defineConfig({
  // ...
  tools: {
    rspack: {
      plugins: [new CompressionPlugin({})]
    }
  }
});
```

**Reply by xc2** [reply]
另外介绍一下这个需求的背景：有的人会希望在打包阶段预先生成好静态资源的 `.gz`, `.br` 文件，在生产环境中直接吞吐这些预先压缩的文件以节省 CPU 资源。

**Reply by mxyzzhong** [reply]
这个插件依赖webpack，如果不想安装webpack貌似用不了

**Reply by xc2** [reply]
webpack 不是必须的，如果集成到 rspack 报错辛苦提个 issue
